### ISSUE TITLE
Argument to controll passing proxy envs from host

### ISSUE THREAD

**Issue by shokerplz** [kubernetes-sigs/kind#3293]
**What would you like to be added**:
An argument like "--no-env-pass" which would allow to not call getProxyEnv in pkg/cluster/internal//providers/docker/provision.go and pkg/cluster/internal//providers/podman/provision.go.

**Why is this needed**:
In my case I have a proxy listening on my local machine and my env vars are looking like HTTPS_PROXY="http://127.0.0.1:3128", so they are being passed to all kind containers and kind clusters doesn't work because of that.
I think it should be possible to disable passing envs.

**Comment by neolit123** [comment]
not a big fan of the --no-foo-bar flag naming.
perhaps a more preferable way would be to have a positive naming such as --foo-bar which is defaulted to true.

in terms of the feature itself, it seems reasonable to me, but perhaps it should be added in the kind config and not as CLI flag.

**Comment by shokerplz** [comment]
yeah, `--pass-env false` is probably better then my initial idea. 
But about config, from what I see, only kubernetes cluster related things (like amount of nodes, cni, etc) should be in config and passing proxy env option is not one of such things.

**Comment by shokerplz** [comment]
yes we can live without this feature, but if kind is automatically grabs those envs from host's shell, why don't we add a feature that allows to disable such behaviour?

**Comment by aojea** [comment]
right now they can do `HTTPS_PROXY="" kind create cluster` and you are suggesting they should do `kind create cluster --pass-env false` 

I can't see why adding a dependency to kind like this will benefit users, since they already have to modify their scripts to set the flag, when they can modify the script to override the env variables

**Comment by BenTheElder** [comment]
The `env` tool or else `HTTPS_PROXY=""` seems more appropriate?

The proxy is an environment detail and it is completely standard behavior to respect these env and expect that they are set reasonably.

**Comment by BenTheElder** [comment]
> I can't see why adding a dependency to kind like this will benefit users, since they already have to modify their scripts to set the flag, when they can modify the script to override the env variables

More importantly, third party scripts and tools should not be setting this flag, for most users respecting these envs is the right thing to do and e.g. test scripts in other projects using kind should not set this flag.

But you can set these env when going to use kind, and then theses other scripts will just work.

We could add a new non-standard env to make kind specifically ignore these, but then .... you could just set the existing standardized env?
